### DISCUSSION TITLE
Wordwrap and "full-width" characters of Chinese/Japanese/Korean

### DISCUSSION THREAD

**Discussion by ksze** [pallets/jinja#1552] [Q&A]
Is the wordwrap filter smart about full-width characters?

Usually a full-width character from the CJK languages is considered to have the width of two ASCII characters.

This is important for wordwrapping plain text.

**Comment by mondeja** [comment]
The wordwrap filter uses Python's `textwrap` library and this does not fit well with Chinese/Japanese/Korean characters. I've seen this bug in so many programs.

The problem is simple but the solution I'm afraid that will not reach soon. It seems that Python is missing a reliable [Unicode Line Breaking Algorithm](https://www.unicode.org/reports/tr14/tr14-45.html) implementation, so the high level developers opt for the [ASCII-oriented `textwrap`, which doesn't have a good unicode support](https://bugs.python.org/issue30680) (there are some variants of the same bug in [bugs.python.org#41975](https://bugs.python.org/issue41975), https://github.com/izimobil/polib/issues/96, https://github.com/AFPy/powrap/issues/76 and https://github.com/mondeja/mdpo/issues/153#issuecomment-879899012).
